scripts/typedef.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * @typedef {import('bn.js')} BN
+ */
+
+/**
+ * Ethereum addresses are composed of the prefix "0x", a common identifier for hexadecimal,
+ * concatenated with the rightmost 20 bytes of the Keccak-256 hash (big endian) of the ECDSA public key
+ * (cf. https://en.wikipedia.org/wiki/Ethereum#Addresses)
+ *
+ * @typedef Address
+ */
+
+/**
+ * Smart contracts are high-level programming abstractions that are compiled down
+ * to EVM bytecode and deployed to the Ethereum blockchain for execution.
+ * This particular type is that of a JS object representing the Smart contract ABI.
+ * (cf. https://en.wikipedia.org/wiki/Ethereum#Smart_contracts)
+ *
+ * @typedef SmartContract
+ */
+
+/**
+ * Example:
+ * {
+ *   amount: 100,
+ *   tokenAddress: 0x0000000000000000000000000000000000000000,
+ *   bracketAddress: 0x0000000000000000000000000000000000000001
+ * }
+ *
+ * @typedef Deposit
+ * @type {object}
+ * @property {number} amount integer denoting amount to be deposited
```

Comment:
This should be a string or a BN.

Reply:
Is it just string or BN? Can it also be number?